EventFilter/L1TRawToDigi/plugins/OmtfUnpacker.cc

```diff
@@ -0,0 +1,368 @@
+    //}
+   
+
+    //
+    // AMC13 trailer
+    //
+    const Word64* trailerAmc13raw = trailer-1;
+    amc13::Trailer trailerAmc13(trailerAmc13raw);
+    if (debug) {
+      std::ostringstream str;
+      str <<" trailerAMC13:  "<<  *reinterpret_cast<const std::bitset<64>*> (trailerAmc13raw) << std::endl;
+      str <<" crc:  "<< trailerAmc13.getCRC() << std::endl;
+      str <<" block: "<< trailerAmc13.getBlock() << std::endl;
+      str <<" LV1ID:  "<< trailerAmc13.getLV1ID() << std::endl;
+      str <<" BX:  "<< trailerAmc13.getBX() << std::endl;
+      LogTrace("") << str.str();
+    }
+
+   
+    //
+    // get AMC13 payload (-> AMC's data)
+    //
+    amc13::Packet packetAmc13;
+    if (!packetAmc13.parse( header, header+1, nWords-2, fedHeader.lvl1ID(), fedHeader.bxID(), true, false)) {
+      edm::LogError("OMTF") << "Could not extract AMC13 Packet.";
+      return;
+    } 
+    //LogTrace("") <<"AMC13 Packet: "<< packetAmc13.blocks() << " size "<<packetAmc13.size() << std::endl;  
+
+
+    //
+    // loop over AMC's
+    //
+    unsigned int blockNum=0;
+    for (auto amc: packetAmc13.payload()) {
+      amc::BlockHeader bh =  amc.blockHeader(); 
+      if (debug) {
+        std::ostringstream str;
+        str <<" ----------- #"<<blockNum++ << std::endl;
+        str <<" blockheader:  "<<  std::bitset<64>(bh.raw()) << std::endl;
+        str <<" boardID:  "<< bh.getBoardID() << std::endl;
+        str <<" amcNumber:  "<< bh.getAMCNumber() << std::endl;
+        str <<" size:  "<< bh.getSize(); // << std::endl;
+        LogTrace("") << str.str();
+      }
+
+      //
+      // AMC header
+      //
+      amc::Header headerAmc = amc.header(); 
+      if (debug) {
+        std::ostringstream str;
+        str <<" AMC header[0]:  "<<  std::bitset<64>(headerAmc.raw()[0]) << std::endl;
+        str <<" AMC header[1]:  "<<  std::bitset<64>(headerAmc.raw()[1]) << std::endl;
+        str <<" AMC number:     "<< headerAmc.getAMCNumber(); // << std::endl;
+        LogTrace("") << str.str();
+      }
+
+
+      //
+      // AMC trailer
+      //
+      //amc::Trailer trailerAmc = amc.trailer();              //this is the expected way but does not work 
+      amc::Trailer trailerAmc(amc.data().get()+amc.size()-1); //FIXME: the above is prefered but this works (CMSSW900)
+      if (debug) {
+        std::ostringstream str;
+        str <<" AMC trailer:  "<<  std::bitset<64>(trailerAmc.raw()) << std::endl;
+        str <<" getLV1ID:     "<< trailerAmc.getLV1ID() << std::endl;
+        str <<" size:         "<< trailerAmc.getSize()  << std::endl;
+        LogTrace("") << str.str();
+      }
+
+      //
+      // AMC payload
+      //
+      const auto & payload64 = amc.data();
+      const Word64* word = payload64.get();
+      for (unsigned int iWord= 1; iWord<= amc.size(); iWord++, word++) {
+        if (iWord<=2 ) continue; // two header words for each AMC
+        if (iWord==amc.size() ) continue; // trailer for each AMC 
```

Comment:
just change the loop to <amc.size() rather than <=

Reply:
It would introduce a change.
 "word" would never reach value payload64.get() + amc.size() in the increment part of the loop.

Reply:
ok - missed that.